### DISCUSSION TITLE
Add a way to hide or restrict access to fields when creating the first user

### DISCUSSION THREAD

**Discussion by joas8211** [payloadcms/payload#2164] [Feature Requests & Ideas]
I'm developing a multi-tenancy plugin for Payload CMS and I'd like to hide the tenant field from the setup form. Read more about the issue here: https://github.com/joas8211/payload-tenancy/issues/1

Currently the form for creating the first user renders all fields in the collection + couble of hardcoded fields. It doesn't check field access, probably because it would fail anyway in most cases.

I can think of a few of possible solutions for this:

1. Add a `field.admin.setup` boolean flag for the field to indicate if it's wanted to be shown at setup.
2. Make `field.admin.hidden` flag a function for conditional rendering.
3. Improve default field access rights and check them at setup form and in the API endpoint it's using.

**Comment by DanRibbens** [comment]
My first thought is that you could use [conditional logic](https://payloadcms.com/docs/fields/overview#conditional-logic) on the field.

```ts
// field
admin: {
  condition: () => !window?.location?.includes('/create-first-user'),
}
```

I haven't tested this yet, so you might need to play with it to get it to work. I would also name the function and assign it where you need for readability and reuse.

Does that solve your issue?

**Reply by joas8211** [reply]
I tried it, and it didn't work. The conditional logic doesn't seem to run on that form either.

**Reply by pcbal** [reply]
Create first user > upload avatar> does that field work before any user is present? When clicked on the upload link a white screen of death is all that is currently available.

**Comment by choutkamartin** [comment]
What @DanRibbens wrote works. However you need to check if `window` is available or not. Here's my snippet:

For my use case the first user page is located at `/admin/create-first-user`.
```ts
{
  name: "image",
  type: "upload",
  relationTo: "media",
  admin: {
    condition: () => {
      if (typeof window !== "undefined") {
        const path = window.location.pathname;
        if (path === "/admin/create-first-user") {
          return false;
        }
        return true
      }
    },
  },
},
```

**Comment by joas8211** [comment]
Conditional rendering worked indeed. I was just being dumb. I somehow managed to mixup the fields and so tried to apply it to a wrong field of a wrong collection. Thanks for helping me @DanRibbens and @choutkamartin!

**Comment by kpc2** [comment]
I also wanted to hide a field on first account creation and came across this discussion. Here's what I ended up doing that worked for me:

```js
admin: {
	condition: (_data, _siblingData, { user }) => Boolean(user),
},
```
